From 992ef3dd1fc0d46b29f20b315adc83fa5f43ee15 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:54 -0400
Subject: [ethernet] mlx4: Fix inaccurate return value of mlx4_flow_attach()

Message-id: <c189a80acb550317b941e14431cb6477c2d9ff0c.1410425016.git.avadai@redhat.com>
Patchwork-id: 93637
O-Subject: [Patch RHEL7.1 V1 118/197] net/mlx4_core: Fix inaccurate return value of mlx4_flow_attach()
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eyal Perry <eyalpe@mellanox.com>

Adopt the "info: why not propagate 'ret' from parse_trans_rule()..."
suggestion made by the smatch semantic checker on:
drivers/net/ethernet/mellanox/mlx4/mcg.c:867 mlx4_flow_attach()

Signed-off-by: Eyal Perry <eyalpe@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 7572038409d0288ff5ba9c77a053ddec6791a43c)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/mcg.c b/drivers/net/ethernet/mellanox/mlx4/mcg.c
index 7c6eba6..4c36def 100644
--- a/drivers/net/ethernet/mellanox/mlx4/mcg.c
+++ b/drivers/net/ethernet/mellanox/mlx4/mcg.c
@@ -897,7 +897,7 @@ int mlx4_flow_attach(struct mlx4_dev *dev,
   ret = parse_trans_rule(dev, cur, mailbox->buf + size);
   if (ret < 0) {
    mlx4_free_cmd_mailbox(dev, mailbox);
-   return -EINVAL;
+   return ret;
   }
   size += ret;
  }
-- 
1.7.1